imports/subscriptions/BoardsSubscriber.ts

```diff
@@ -100,4 +108,53 @@ class BoardsSubscriberClass {
   }
 }
 
+class BoardFavoritesSubscriberClass {
+  private _isSubscribed = false
+
+  public subscribe() {
+    if (!this._isSubscribed) {
+      this._isSubscribed = true
+      this._subscribe()
+    }
+  }
+
+  private _subscribe() {
+    Meteor.subscribe(Collections.names.board_favorites)
+    const BoardFavoritesCollection = Collections.getCollectionByName('board_favorites')
+    const onFavoritesChanged = this._onFavoritesChanged.bind(this)
+    BoardFavoritesCollection.find({}).observeChanges({
+      added(_id, _fields) {
+        onFavoritesChanged()
+      },
+      changed(_id, _fields) {
+        onFavoritesChanged()
+      },
+      removed(_id) {
+        onFavoritesChanged()
+      }
+    })
+  }
+
+  private async _onFavoritesChanged() {
+    const favoriteBoards = await this.getFavoriteBoards()
+    globalEventEmitter.emit(BoardEvents.FavoritesChanged, favoriteBoards)
+  }
+
+  public async getFavoriteBoards(): Promise<IBoard[]> {
```

Comment:
1. do not use `const userid = Meteor.userId()! `on client
2. do not use `collection` in client

wrap all of the them into `service`,  see how we do it in `BoardService`